CMakeLists.txt

```diff
@@ -122,14 +125,34 @@ else()
 		set_cxx_flag("-O3" RELWITHDEBINFO)
 	endif()
 
+	try_cxx_flag(FNO_MATH_ERRNO "-fno-math-errno")
+
 	if (USE_FAST_MATH)
-		# By default, GCC uses -ffp-contract=fast with -std=gnu* and uses -ffp-contract=off with -std=c*.
-		# See https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html
-		# By default, GCC doesn't enable the -ffast-math option.
-		set_cxx_flag("-ffast-math -fno-math-errno -ffp-contract=fast")
+		try_cxx_flag(FFAST_MATH "-ffast-math")
+
+		# GCC.
+		try_cxx_flag(FFP_CONTRACT_FAST "-ffp-contract=fast")
+		# Clang.
+		try_cxx_flag(FFP_MODEL_FAST "-ffp-model=agressive")
+		# ICC.
+		try_cxx_flag(FP_MODEL_FAST_2 "-fp-model=fast=2")
+		try_cxx_flag(QSIMD_HONOR_FP_MODEL "-qsimd-honor-fp-model")
 	else()
+		try_cxx_flag(FNO_FAST_MATH "-fno-fast-math")
+
+		# By default, GCC uses -ffp-contract=fast with -std=gnu* and uses -ffp-contract=off with -std=c*.
 		# By default, GCC uses -std=gnu* and then enables -ffp-contract=fast even if -ffast-math is not enabled.
-		set_cxx_flag("-ffp-contract=off")
+		# See https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html
+		# GCC fast contractions (-ffp-contract=fast) should be safe, but aren't on arm64 with GCC 12.
```

Comment:
What do you mean by "aren't safe"?

Reply:
I should have been more verbose at the time because I forgot the exact issue.

I guess it meant it doesn't produce the same result as compiling without any fast stuff.

Reply:
Given the purpose of my patch is to maximize the chance the files are reproducible, I probably noticed that on ARM such option broke the reproducibility. It's probably a similar problem than using x87 instead of SSE on x86, maybe some ARM fused operations break IEEE compliance.

Reply:
For sure, since I mentioned a specific GCC version, that was the result of me testing that specific compiler on the said hardware, and I was testing the reproducibility of converted files. GCC 12 is the Debian Bookworm GCC, and I use Debian Bookworm on my Arm boards.

Reply:
That's unobvious so it should be explained in the comment.

As we've discussed in the past, I don't think floating point reproducibility is a good goal -- the language and compilers don't make any attempt to provide such guarantees. Platforms using x87 floating point are an easy example where you're not going to achieve it.  But for GCC those options to disable fast math are good in any case, since fast math makes the software too unreliable.